### PR TITLE
Reverse string solution 

### PR DESCRIPTION
**Pull request by Mo-Altayeb**
---
name: Reverse Challenge solution and test files
---

## Behavior
This function takes a string as input and returns a string that contains the contents of the input string in reversed order. Related to issue #27 
### Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations
- [ ] The function is not called at the top level of the function file 
  - _Recursive solutions **can** call the function from **inside** the function body_

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] There are no commented lines of code
- [ ] There are no `print` statements anywhere
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/tests/test_reverse_string.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    def test_normal_input_numbers(self):
+        """it should return a string with the numbers in reversed order"""
+        self.assertEqual(reverse_string("123456"), "654321")
+
+    def test_capital_letters_input(self):
+        """it should reverse the string without affecting capitalization"""
+        self.assertEqual(reverse_string("AbscEfgh"), "hgfEcsbA")
+
+    # Edge cases
+    def test_mixed_numbers_letters_input(self):
```
Comment (Razan-O-Elobeid) [inline solutions/tests/test_reverse_string.py:38]: why do you consider it n edge case?
but also i suggest to rename it as **"test_string_with_mixed_numbers_and_letters"**

File: solutions/tests/test_reverse_string.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        self.assertEqual(
+            reverse_string("12345678910111213141516171819"),
+            "91817161514131211101987654321",
+        )
+
+    def test_empty_string(self):
+        """it should return an empty string"""
+        self.assertEqual(reverse_string(""), "")
+
+    def test_blank_space_input(self):
```
Comment (Razan-O-Elobeid) [inline solutions/tests/test_reverse_string.py:53]: Do you mean a sentence? I suggest placing it under standard tests and renaming it to **test_string_with_sentence**

**Review by Razan-O-Elobeid — CHANGES_REQUESTED** [review]
Revise the test function names to make them more descriptive and aligned with the specific inputs being tested

File: solutions/tests/test_reverse_string.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import unittest
+
+from ..reverse_string import reverse_string
+
+
+class TestReverseString(unittest.TestCase):
+    """Test suite for the reverse_string function for reversing an input string"""
+
+    # Standard cases
+    def test_normal_input_letters(self):
```
[the remark this answers is not in the document]
Comment (Mo-Altayeb) [reply solutions/tests/test_reverse_string.py:25]: Yes you are right, I just wanted the test to be sort of a general test that includes different types of inputs, but I guess that's not a good way to go with unit tests. I will change them as soon as I get the chance.

File: solutions/tests/test_reverse_string.py (hunk 2 above)
Comment (Mo-Altayeb) [reply solutions/tests/test_reverse_string.py:53]: Not necessarily a sentence, but blank spaces in general (could be spaces between numbers or letters)

**Comment by Razan-O-Elobeid** [comment]
Good Job Mohamed
